src/components/date-picker/date-picker.ts

```diff
@@ -0,0 +1,792 @@
+  }
+
+  protected handleDialogClosing(event: Event) {
+    event.stopPropagation();
+    this._hide(true);
+  }
+
+  protected handleDialogClosed(event: Event) {
+    event.stopPropagation();
+  }
+
+  private setDateConstraints() {
+    const dates: DateRangeDescriptor[] = [];
+    if (this._min) {
+      dates.push({ type: DateRangeType.Before, dateRange: [this._min] });
+    }
+    if (this._max) {
+      dates.push({ type: DateRangeType.After, dateRange: [this._max] });
+    }
+    if (this._disabledDates?.length) {
+      dates.push(...this.disabledDates);
+    }
+
+    this._dateConstraints = dates.length ? dates : undefined;
+  }
+
+  private renderClearIcon() {
+    return !this.value
+      ? nothing
+      : html`
+          <span slot="suffix" part="clear-icon" @click=${this.clear}>
+            <slot name="clear-icon">
+              <igc-icon
+                name="clear"
+                collection="internal"
+                aria-hidden="true"
+              ></igc-icon>
+            </slot>
+          </span>
+        `;
+  }
+
+  private renderCalendarIcon() {
+    const defaultIcon = html`
+      <igc-icon
+        name="calendar_today"
+        collection="internal"
+        aria-hidden="true"
+      ></igc-icon>
+    `;
+
+    const state = this.open ? 'calendar-icon-open' : 'calendar-icon';
+
+    return html`
+      <span slot="prefix" part=${state} @click=${this.handleAnchorClick}>
+        <slot name=${state}>${defaultIcon}</slot>
+      </span>
+    `;
+  }
+
+  private renderCalendar(id: string) {
+    const hideHeader = this.isDropDown ? true : this.hideHeader;
+
+    return html`
+      <igc-calendar
+        aria-labelledby=${id}
+        role="dialog"
+        .inert=${!this.open || this.disabled}
+        ?show-week-numbers=${this.showWeekNumbers}
+        ?hide-outside-days=${this.hideOutsideDays}
+        ?hide-header=${hideHeader}
+        .activeDate=${this.activeDate ?? this.value ?? new Date()}
+        .value=${this.value ?? undefined}
+        .headerOrientation=${this.headerOrientation}
+        .orientation=${this.orientation}
+        .visibleMonths=${this.visibleMonths}
+        .locale=${this.locale}
+        .disabledDates=${this._dateConstraints!}
+        .specialDates=${this.specialDates}
+        .weekStart=${this.weekStart}
+        @igcChange=${this.handleCalendarChangeEvent}
+        exportparts="header, header-title, header-date, content: calendar-content, navigation, months-navigation,
+        years-navigation, years-range, navigation-buttons, navigation-button, days-view-container,
+        days-view, months-view, years-view, days-row, label: calendar-label, week-number, week-number-inner, date,
+        date-inner, first, last, inactive, hidden, weekend, range, special, disabled, single, preview,
+        month, month-inner, year, year-inner, selected, current"
+      >
+        ${!this.isDropDown
+          ? html`<slot name="title" slot="title">Select date</slot> `
```

Comment:
This is already the default title from the resource strings from what I can see, so passing this slot along... only disables the resource string handling? I think should be removed if so.